EIPS/eip-5750.md

```diff
@@ -8,103 +8,91 @@ status: Review
-
-Standardizing how method will be extended helps answer this questions.
-
-In particular, this EIP aims at achieving maximum backward (e.g. [EIP-721](./eip-721.md), [EIP-1155](./eip-1155.md)) and future compatibility e.g. supporting commit-reveal feature for arbitrary method ([EIP-5732](./eip-5732.md)), or adding digital signatures alongside with a method call. The goal and motivation is to allow other implementers and EIPs to further depend on the compatibility based on the specifications this EIP so all compliant method interface automatically becomes eligible for future new behaviors.
+Finally, this EIP aims to achievie maximum backward and future compatibility. Many EIPs already partially support this EIP, such as [EIP-721](./eip-721.md) and [EIP-1155]. This EIP supports many use cases, from commit-reveal schemes ([EIP-5732](./eip-5732.md)), to adding digital signatures alongside with a method call. Other implementers and EIPs should be able to depend on the compatibility granted by this EIP so that all compliant method interfaces are eligible for future new behaviors.
 
 ## Specification
 
-The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119 and RFC 8174.
 
-1. Any compliant contract's compliant method MUST have a `bytes` (dynamic size) as its last parameter of the method.
+A method MUST be compliant if and only if for all of its overloads, it already has a `bytes` (dynamic size) as the last parameter that is not used for any functionality, or has another overload with the exact same parameters and a `bytes` (dynamic size) as the last parameter that is not used for any functionality:
 
 ```solidity
-function methodName(type1 param1, type2, param2 ... bytes calldata data);
+// `methodName1` is compliant: the first overload has a similar overload (the second overload) with the bytes parameter, and the second overload has the bytes parameter
+function methodName1(uint256 param1, address param2);
+function methodName1(uint256 param1, address param2, bytes calldata data);
+// `methodName2` is compliant: its only overload has the bytes parameter
+function methodName2(address param1, address param2, bytes calldata data);
+// `methodName3` is not compliant: it has an overload (the third one) that doesn't have a valid overload
+function methodName3(string calldata param1, bytes32 param2);
+function methodName3(string calldata param1, bytes32 param2, bytes calldata data);
+function methodName3(string calldata param3, address param4);
 ```
 
+For the purposes of calculating [EIP-165](./eip-165.md) interface identifiers, the overloads created for the purpose of compliance with this EIP MUST be considered. 
```

Comment:
I am a little hesitant of adding the EIP-165 mentioning but we continue to discuss and address it in a follow up commit

Reply:
The reason I included it was because I was initially considering making it a "MUST NOT" instead, since that seemed like the best choice. Then I realized that would break compatibility with EIP-165.

This could be moved to the Backward Compatibility section, if you prefer.

Reply:
I think we need to fix this Markdown Linter erro for Line 66 @Pandapip1